lib/secretsscanner/scan/scan_test.go

```diff
@@ -0,0 +1,192 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package scan
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+	cryptosshtestdata "golang.org/x/crypto/ssh/testdata"
```

Comment:
Is it wise to import test data from a third party package?

This seems like the kind of package that could easily change or disappear during a version update, leaving us broken.

It also seems like our test scenarios could silently shift without us realizing it.

Reply:
I attempted to avoid it since Panther would flag a secret leak in this PR.

Anyway, I forked the ssh's test data private keys, retained the credits, and we can proceed with it.